libs/shared/common/src/lib/utilities/form.utils.ts

```diff
@@ -0,0 +1,34 @@
+import { AbstractControl, FormGroup } from '@angular/forms';
+import { FormControlExtras, FormGroupTypeGenerator } from '@expenses-tracker/shared/interfaces';
+
+import { camelCaseToSentenceCase } from './string.utils';
+
+const getError = (control: AbstractControl<unknown, unknown>): boolean => {
+  return control.touched && !!control.errors;
+};
+
+const getErrorMessage = <C extends { [key: string]: unknown }, F extends FormGroup<FormGroupTypeGenerator<C>>>(
+  formGroup: F,
+  formControlName: keyof C
+) => {
+  const control = camelCaseToSentenceCase(formControlName as string);
+  if (formGroup['controls'][formControlName]?.hasError('email')) {
+    return `Invalid ${control.toLowerCase()} address`;
+  }
+  if (formGroup['controls'][formControlName]?.hasError('required')) {
+    return `${control} is required`;
+  }
+  return `Unknown validation error for ${control}`;
+};
+
+const controlStateValidator = <C extends { [key: string]: unknown }, F extends FormGroup<FormGroupTypeGenerator<C>>>(

```

Comment:
Add documentation

Reply:
Resolved in https://github.com/usamazansari/expenses-tracker/pull/62/commits/23dcbcd6064df9bcf23c155f20386b21e0cb95f8